kedro-datasets/kedro_datasets/snowflake/snowpark_dataset.py

```diff
@@ -0,0 +1,224 @@
+"""``AbstractDataSet`` implementation to access Snowflake using Snowpark dataframes
+"""
+import logging
+from copy import deepcopy
+from typing import Any, Dict, Union
+
+import pandas as pd
+import snowflake.snowpark as sp
+from kedro.io.core import AbstractDataSet, DataSetError
+
+logger = logging.getLogger(__name__)
+
+
+class SnowparkDataSet(AbstractDataSet[pd.DataFrame, pd.DataFrame]):
+    """``SnowparkDataSet`` loads and saves Snowpark dataframes.
+
+    Example usage for the
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+    data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
+
+    .. code-block:: yaml
+        weather:
+          type: kedro_datasets.snowflake.SnowparkDataSet
+          table_name: "weather_data"
+          database: "meteorology"
+          schema: "observations"
+          credentials: db_credentials
+          save_args:
+            mode: overwrite
+            column_order: name
+            table_type: ''
+
+    One can skip everything but "table_name" if database and
+    schema provided via credentials. Therefore catalog entries can be shorter
+    if ex. all used Snowflake tables live in same database/schema.
+    Values in dataset definition take priority over ones defined in credentials
+
+    Example:
+    Credentials file provides all connection attributes, catalog entry
+    "weather" reuse credentials parameters, "polygons" catalog entry reuse
+    all credentials parameters except providing different schema name
+
+    catalog.yml
+
+    .. code-block:: yaml
+        weather:
+          type: kedro_datasets.snowflake.SnowparkDataSet
+          table_name: "weather_data"
```

Comment:
What about loading a DataSet from a query instead of a Table

Reply:
Current scope was only for tables. We can consider adding query-based datasets in next increment. That will require some thinking, as query-based dataset will be read-only